test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ActionOverloadingTests/Tests/ActionOverloadingEndToEndTests.cs

```diff
@@ -0,0 +1,266 @@
+                    "POST",
+                    new BodyOperationParameter("n", 3));
+
+                // action bound with collection of special employees
+                _context.Execute(
+                    new Uri(_baseUri + "People/Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.SpecialEmployee/Default.IncreaseSalaries", UriKind.Absolute),
+                    "POST",
+                    new BodyOperationParameter("n", 3));
+            }
+        }
+
+        [Fact]
+        public void OverloadedActionsProjection()
+        {
+            for (int i = 1; i < 2; i++)
+            {
+                Common.Clients.EndToEnd.Product product = _context.Execute<Common.Clients.EndToEnd.Product>(new Uri(_baseUri + "Products(-10)?$select=Default.*", UriKind.Absolute)).Single();
+                OperationDescriptor productOperationDescriptor = _context.GetEntityDescriptor(product).OperationDescriptors.Single();
+                Assert.Equal(_baseUri + "$metadata#Default.RetrieveProduct", productOperationDescriptor.Metadata.AbsoluteUri, true);
+                Assert.Equal(_baseUri + "Products(-10)/Default.RetrieveProduct", productOperationDescriptor.Target.AbsoluteUri, true);
+
+                Common.Clients.EndToEnd.OrderLine orderLine = _context.Execute<Common.Clients.EndToEnd.OrderLine>(new Uri(_baseUri + "OrderLines(OrderId=-10,ProductId=-10)?$select=*")).Single();
+                Assert.Empty(_context.GetEntityDescriptor(orderLine).OperationDescriptors);
+            }
+        }
+
+        private void VerifyLinks(Dictionary<string, string> expectedValues, IEnumerable<OperationDescriptor> actualDescriptors)
+        {
+            foreach (KeyValuePair<string, string> expected in expectedValues)
+            {
+                OperationDescriptor od;
+                od = actualDescriptors.Where(d => d.Metadata.AbsoluteUri.Equals(_baseUri + expected.Key, StringComparison.OrdinalIgnoreCase)).First();
+                Assert.Equal(_baseUri + expected.Value, od.Target.AbsoluteUri, true);
+            }
+        }
+
+        private void VerifyLinks(IEnumerable<Tuple<string, string>> expectedValues, IEnumerable<OperationDescriptor> actualDescriptors)
+        {
+            foreach (var expected in expectedValues)
+            {
+                IEnumerable<OperationDescriptor> ods;
+                ods = actualDescriptors.Where(d => d.Metadata.AbsoluteUri.Equals(_baseUri + expected.Item1, StringComparison.OrdinalIgnoreCase));
+
+                bool matched = false;
+                foreach (var od in ods)
+                {
+                    if (_baseUri + expected.Item2 == od.Target.AbsoluteUri)
+                    {
+                        matched = true;
+                        break;
+                    }
+                }
+
+                Assert.True(matched);
+            }
+        }

```

Comment:
Are there tests where you test the return values of actions?

Reply:
Most of these tests were written to show that the actions execute but I've added some asserts for the various actions.